### PR TITLE
overhaul core resource tests

### PR DESCRIPTION
**Pull request by codycushing**
* fix defects in update functions for drg, drgAttachment, ipsecConnection, subnet and vcn
* fix internet gateway "enabled" bug
* handle additional states for console history resource
* consistent convention based naming
* convert to resource.Test for all acceptance testing
* combine create and update into multistep tests
* remove superfluous delete tests
* destructive change example test approach in ResourceCoreVirtualNetworkTestSuite
* non TestSuite based approach in TestAccResourceCoreSubnetCreate_basic

### Test output
$ TF_ORACLE_ENV=test TF_ACC=1 go test -v -timeout 120m -run TestResourceCore
=== RUN   TestResourceCoreConsoleHistoryTestSuite
=== RUN   TestAccResourceCoreInstanceConsoleHistory_basic
--- PASS: TestAccResourceCoreInstanceConsoleHistory_basic (168.90s)
--- PASS: TestResourceCoreConsoleHistoryTestSuite (168.90s)
=== RUN   TestResourceCoreCpeTestSuite
=== RUN   TestAccResourceCoreCpe_basic
--- PASS: TestAccResourceCoreCpe_basic (0.50s)
--- PASS: TestResourceCoreCpeTestSuite (0.50s)
=== RUN   TestResourceCoreDHCPOptionsTestSuite
=== RUN   TestAccResourceCoreDHCPOptions_basic
--- PASS: TestAccResourceCoreDHCPOptions_basic (8.98s)
--- PASS: TestResourceCoreDHCPOptionsTestSuite (8.98s)
=== RUN   TestResourceCoreDrgAttachmentTestSuite
=== RUN   TestAccResourceCoreDrgAttachment_basic
--- PASS: TestAccResourceCoreDrgAttachment_basic (122.92s)
--- PASS: TestResourceCoreDrgAttachmentTestSuite (122.92s)
=== RUN   TestResourceCoreDrgTestSuite
=== RUN   TestAccResourceCoreDrg_basic
--- PASS: TestAccResourceCoreDrg_basic (138.53s)
--- PASS: TestResourceCoreDrgTestSuite (138.53s)
=== RUN   TestResourceCoreImageTestSuite
=== RUN   TestAccCoreImage_basic
--- PASS: TestAccCoreImage_basic (938.40s)
--- PASS: TestResourceCoreImageTestSuite (938.41s)
=== RUN   TestResourceCoreInstanceTestSuite
=== RUN   TestAccResourceCoreInstance_basic
--- PASS: TestAccResourceCoreInstance_basic (166.05s)
--- PASS: TestResourceCoreInstanceTestSuite (166.05s)
=== RUN   TestResourceCoreInternetGatewayTestSuite
=== RUN   TestAccResourceCoreInternetGateway_basic
--- PASS: TestAccResourceCoreInternetGateway_basic (8.25s)
--- PASS: TestResourceCoreInternetGatewayTestSuite (8.25s)
=== RUN   TestResourceCoreIPSecTestSuite
=== RUN   TestAccResourceCoreIpsec_basic
--- PASS: TestAccResourceCoreIpsec_basic (438.02s)
--- PASS: TestResourceCoreIPSecTestSuite (438.02s)
=== RUN   TestResourceCoreRouteTableTestSuite
=== RUN   TestAccResourceCoreRouteTable_basic
--- PASS: TestAccResourceCoreRouteTable_basic (42.55s)
--- PASS: TestResourceCoreRouteTableTestSuite (42.56s)
=== RUN   TestResourceCoreSecurityListTestSuite
=== RUN   TestAccResourceCoreSecurityList_basic
--- PASS: TestAccResourceCoreSecurityList_basic (10.85s)
--- PASS: TestResourceCoreSecurityListTestSuite (10.86s)
=== RUN   TestResourceCoreVirtualNetworkTestSuite
=== RUN   TestAccResourceCoreVirtualNetwork_basic
--- PASS: TestAccResourceCoreVirtualNetwork_basic (8.34s)
--- PASS: TestResourceCoreVirtualNetworkTestSuite (8.34s)
=== RUN   TestResourceCoreVnicAttachmentTestSuite
=== RUN   TestAccCoreVnicAttach_basic
--- PASS: TestAccCoreVnicAttach_basic (230.12s)
--- PASS: TestResourceCoreVnicAttachmentTestSuite (230.13s)
=== RUN   TestResourceCoreVolumeAttachmentTestSuite
=== RUN   TestResourceCoreVolumeAttachment_basic
--- PASS: TestResourceCoreVolumeAttachment_basic (203.32s)
--- PASS: TestResourceCoreVolumeAttachmentTestSuite (203.32s)
=== RUN   TestResourceCoreVolumeBackupTestSuite
=== RUN   TestCreateVolumeBackup
--- PASS: TestCreateVolumeBackup (62.54s)
--- PASS: TestResourceCoreVolumeBackupTestSuite (62.54s)
=== RUN   TestResourceCoreVolumeTestSuite
=== RUN   TestCreateResourceCoreVolume
--- PASS: TestCreateResourceCoreVolume (28.27s)
--- PASS: TestResourceCoreVolumeTestSuite (28.27s)
PASS
ok      github.com/oracle/terraform-provider-oci        2576.590s

### CODE REVIEW

File: resource_obmcs_core_internet_gateway.go [hunk 1]
```diff
@@ -144,9 +144,14 @@ func (s *InternetGatewayResourceCrud) Get() (e error) {
 
 func (s *InternetGatewayResourceCrud) Update() (e error) {
 	opts := &baremetal.UpdateGatewayOptions{}
-	if isEnabled, ok := s.D.GetOk("enabled"); ok {
-		opts.IsEnabled = isEnabled.(bool)
-	}
+
+	// todo: GetOk malfunction with this bool: 'ok' is always the value of the bool
```
Comment (briangustafson) [inline resource_obmcs_core_internet_gateway.go]: Is this maybe https://github.com/hashicorp/terraform/issues/13512? We work around that in at least one other place.

File: resource_obmcs_core_virtual_network_test.go [hunk 2]
```diff
@@ -18,137 +17,79 @@ type ResourceCoreVirtualNetworkTestSuite struct {
...
+					resource "oci_core_virtual_network" "t" {
+						cidr_block = "10.0.0.0/24"
+						compartment_id = "${var.compartment_id}"
+					}`,
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(s.ResourceName, "cidr_block", "10.0.0.0/24"),
+					func(s *terraform.State) (err error) {
+						resId2, err := fromInstanceState(s, "oci_core_virtual_network.t", "id")
+						if resId == resId2 {
+							return fmt.Errorf("Expected new vcn ocid, got the same")
```
Comment (briangustafson) [inline resource_obmcs_core_virtual_network_test.go]: neat

File: vendor/github.com/MustWin/baremetal-sdk-go/request_options.go [hunk 3]
```diff
@@ -259,7 +259,7 @@ type UpdateDHCPDNSOptions struct {
 type UpdateGatewayOptions struct {
 	IfMatchOptions
 	DisplayNameOptions
-	IsEnabled bool `header:"-" json:"isEnabled,omitempty" url:"-"`
+	IsEnabled *bool `header:"-" json:"isEnabled,omitempty" url:"-"`
```
Comment (briangustafson) [inline vendor/github.com/MustWin/baremetal-sdk-go/request_options.go]: Should this change be in here?

File: resource_obmcs_core_internet_gateway.go (hunk 1 above)
Comment (codycushing) [reply resource_obmcs_core_internet_gateway.go]: I don't think it's that, this has to do with how they verify zero values and do deep compare

File: vendor/github.com/MustWin/baremetal-sdk-go/request_options.go (hunk 3 above)
Comment (codycushing) [reply vendor/github.com/MustWin/baremetal-sdk-go/request_options.go]: Yes, booleans values can only mean true or false, but boolean pointers can mean true, false or nil. True or false need to be transmitted as named json fields in the request, nil needs to be omitted from request so the current or default value is unaltered.

**Pull request merged**